app/views/helpers/dateFields.scala.html

```diff
@@ -45,7 +45,7 @@
         item =>
             InputItem(
                 id = s"${id.getOrElse(fieldName).replace(".", "-")}-$item",
-                classes = s"govuk-input--width-${if(item == "year") "4" else "2"} ${if(form.hasErrors) "govuk-input--error"}",
+                classes = s"govuk-input--width-${if(item == "year") "4" else "2"} ${if(!dateErrors.isEmpty) "govuk-input--error"}",
```

Comment:
can use `nonEmpty` ??? - just a suggestion

Reply:
changed to if(dateErrors. nonEmpty)